### ISSUE TITLE
[display] import diagnostics should take type parameters into account

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#8044]
In this case, a type whose type parameters don't fit is suggested as the preferred choice, so applying the quick fix is then followed by an "invalid number of type parameters" error. This is especially annoying because of VSCode's new "auto fix" command / preferred code action API.

![](https://i.imgur.com/CvaVHc6.png)

If possible, type parameters should be taken into account to sort accordingly from "best" to "worst" fit.

**Comment by Simn** [comment]
I'm reading this and don't really know what you actually mean by "should take type parameters into account". It also doesn't help that I can't tell what you're doing in that screenshot. Could you rephrase the problem here so even I can understand it?

**Comment by Aurel300** [comment]
@Simn There are two types that could be imported: `haxe.format.JsonParser` and `json2object.JsonParser`. The former takes no types parameters, the latter takes one. I think the suggestion is that in this case the latter should be suggested first, because it would work with the type parameters (at least by arity, if not possible constraints).

**Comment by Gama11** [comment]
Exactly, `haxe.format.JsonParser` is not a very good fit because it has _no_ type parameters, whereas the syntax in the source file has _one_.
